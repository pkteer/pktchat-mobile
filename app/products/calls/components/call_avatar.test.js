// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import CallAvatar from './call_avatar';

describe('CallAvatar', () => {
    const baseProps = {
        userId: 'user-id',
        volume: 1,
        muted: false,
        size: 'm',
    };

    test('should match snapshot unmuted', () => {
        const wrapper = shallow(<CallAvatar {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot muted', () => {
        const props = {...baseProps, muted: true};
        const wrapper = shallow(<CallAvatar {...props}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot size large', () => {
        const props = {...baseProps, size: 'l'};
        const wrapper = shallow(<CallAvatar {...props}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
