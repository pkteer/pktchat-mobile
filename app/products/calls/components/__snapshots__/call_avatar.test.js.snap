// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallAvatar should match snapshot muted 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginBottom": 5,
        "width": 40,
      },
      Object {
        "shadowColor": "rgb(61, 184, 135)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 10,
      },
    ]
  }
>
  <Connect(ProfilePicture)
    showStatus={false}
    size={40}
    userId="user-id"
  />
  <CompassIcon
    name="microphone-off"
    size={12}
    style={
      Object {
        "backgroundColor": "black",
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 2,
        "bottom": -5,
        "color": "white",
        "height": 20,
        "overflow": "hidden",
        "padding": 2,
        "position": "absolute",
        "right": -5,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 20,
      }
    }
  />
</View>
`;

exports[`CallAvatar should match snapshot size large 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 36,
        "height": 72,
        "marginBottom": 5,
        "width": 72,
      },
      Object {
        "shadowColor": "rgb(61, 184, 135)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 10,
      },
    ]
  }
>
  <Connect(ProfilePicture)
    showStatus={false}
    size={72}
    userId="user-id"
  />
  <CompassIcon
    name="microphone"
    size={16}
    style={
      Object {
        "backgroundColor": "#3DB887",
        "borderColor": "black",
        "borderRadius": 12,
        "borderWidth": 2,
        "bottom": -5,
        "color": "white",
        "height": 24,
        "overflow": "hidden",
        "padding": 2,
        "position": "absolute",
        "right": -5,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 24,
      }
    }
  />
</View>
`;

exports[`CallAvatar should match snapshot unmuted 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginBottom": 5,
        "width": 40,
      },
      Object {
        "shadowColor": "rgb(61, 184, 135)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 10,
      },
    ]
  }
>
  <Connect(ProfilePicture)
    showStatus={false}
    size={40}
    userId="user-id"
  />
  <CompassIcon
    name="microphone"
    size={12}
    style={
      Object {
        "backgroundColor": "#3DB887",
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 2,
        "bottom": -5,
        "color": "white",
        "height": 20,
        "overflow": "hidden",
        "padding": 2,
        "position": "absolute",
        "right": -5,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 20,
      }
    }
  />
</View>
`;
